### PR TITLE
Fix various copy to clipboard issues

### PR DESCRIPTION
**Pull request by matthiasgeiger**
Pt 1: Fixing #5263 by only using the text content to be added in the right-click-copy functionality in the preview (see https://github.com/JabRef/jabref/pull/5340/files#diff-b84aeae22f7f719bc60f1ae34ecb39f6)

Pt 2: Right click option in maintable: 
Additionally store content also in String format that can be accessed then with simple text editors.

Pt 3: improve also functionality in "Export to clipboard" dialog
Currently all "exporters" are used for "Export to Clipboard" - however, the ones which are creating files like PDFs cannot be used for copying to clipboard - thus they are filtered out now. Additionally, in case RTF or HTML content is exported the specific ClipboardContent has been used.

----

- [x] Change in CHANGELOG.md described
- [x] Tests ~~created~~ adjusted for changes 
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java [hunk 1]
```diff
@@ -115,12 +120,20 @@ private String exportToClipboard(Exporter exporter) {
                 }
             }
         }
-        return "";
+        throw new Exception("export failed.");
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java]: I think then we should also remove the `Exception` handler above (or rethrow there) to get a more detailed exception message of what went wrong.

File: src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java [hunk 2]
```diff
@@ -135,4 +148,14 @@ private String readFileToString(Path tmp) throws IOException {
             return reader.lines().collect(Collectors.joining(OS.NEWLINE));
         }
     }
+
+    private class ExportResult {
+        final String content;
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java]: not sure about java conventions, but shouldn't be variables in a class always be hidden by get/set methods?

Comment (matthiasgeiger) [reply src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java]: More or less, yes. However, as this is a private inner class you can omit it.

File: src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java (hunk 1 above)
Comment (matthiasgeiger) [reply src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java]: As the Exception is already logged above, it's not necessary to get a more detailed message - however, I added the `onFailure()` to swallow the rethrow in 2036ec2

**Comment by matthiasgeiger** [comment]
@tobiasdiez If you don't have further objections, I would merge it to master 😃

**Pull request merged**